### PR TITLE
Consolidate module parameter handling

### PR DESCRIPTION
**Pull request by Pessimistress**
General code quality issues encountered as I'm working on custom shader modules.

#### Change List
- Add `layer.setModuleParameters` method
- Remove direct `model.updateModuleSettings` calls from `drawLayer` and `pickLayer`

### CODE REVIEW

File: examples/layer-browser/src/components/color-picker.js [hunk 1]
```diff
@@ -42,7 +42,8 @@ export function getColorArray(str) {
   return [
     parseInt(str.slice(1, 3), 16),
     parseInt(str.slice(3, 5), 16),
-    parseInt(str.slice(5, 7), 16)
+    parseInt(str.slice(5, 7), 16),
+    255
```
Comment (Pessimistress) [inline examples/layer-browser/src/components/color-picker.js]: `picking` module throws error if `pickingHighlightColor` is array of 3

File: modules/core/src/lib/draw-layers.js [hunk 2]
```diff
@@ -301,18 +301,15 @@ function getViewportFromDescriptor(viewportOrDescriptor) {
...
-  if (Number.isInteger(layer.props.highlightedObjectIndex)) {
-    const pickingSelectedColor =
-      layer.props.highlightedObjectIndex >= 0
-        ? layer.encodePickingColor(layer.props.highlightedObjectIndex)
-        : null;
-
-    return {
-      pickingSelectedColor
-    };
+  if (Number.isInteger(highlightedObjectIndex) && highlightedObjectIndex >= 0) {
```
Comment (ibgreen) [inline modules/core/src/lib/draw-layers.js]: This setting does not need to be cleared if the `highlightedObjectIndex` is not a number? Maybe worth a comment?

File: modules/core/src/lib/layer-state.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ export default class LayerState extends ComponentState {
     this.model = null;
     this.needsRedraw = true;
     this.subLayers = null; // reference to sublayers rendered in a previous cycle
+    this.moduleParameters = {};
```
Comment (ibgreen) [inline modules/core/src/lib/layer-state.js]: The original design of the module parameters system was that one would be able to only supply any new props, and only those uniforms would be generated, avoiding the setting of lots of uniforms. 

Storing and updating a big settings object would seem to totally defeat that.

But perhaps the original `moduleSettings` design doesn't work that well, I recall running into a few issues.

File: modules/core/src/lib/layer.js [hunk 4]
```diff
@@ -662,6 +664,13 @@ ${flags.viewportChanged ? 'viewport' : ''}\
     return this.setChangeFlags(changeFlags);
   }
 
+  setModuleParameters(moduleParameters) {
+    if (this.internalState) {
+      Object.assign(this.internalState.moduleParameters, moduleParameters);
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js]: Seeing it like this does seem to duplicate the intention of the system. If the app has to do this anyway, shouldn't we just move this into the module system?

**Pull request merged**